pkg/reconciler/apis/permissionclaimlabel/permissionclaimlabel_controller.go

```diff
@@ -0,0 +1,257 @@
+			list, err := apiBindingInformer.Lister().List(labels.Everything())
+			if err != nil {
+				return nil, err
+			}
+
+			var ret []*apisv1alpha1.APIBinding
+
+			for i := range list {
+				if logicalcluster.From(list[i]) != clusterName {
+					continue
+				}
+
+				ret = append(ret, list[i])
+			}
+
+			return ret, nil
+		},
+		apiBindingsIndexer: apiBindingInformer.Informer().GetIndexer(),
+	}
+
+	apiBindingInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) { c.enqueueAPIBinding(obj, "") },
+		UpdateFunc: func(_, newObj interface{}) {
+			// TODO: In the future, when we should calculate if there is a need for an update.
+			c.enqueueAPIBinding(newObj, "")
+		},
+		DeleteFunc: func(obj interface{}) { c.enqueueAPIBinding(obj, "") },
+	})
+
+	c.ddsif.AddEventHandler(informer.GVREventHandlerFuncs{
+		AddFunc:    func(gvr schema.GroupVersionResource, obj interface{}) { c.enqueueBindingForResource(gvr, obj) },
+		UpdateFunc: func(gvr schema.GroupVersionResource, _, obj interface{}) { c.enqueueBindingForResource(gvr, obj) },
+		DeleteFunc: nil, // Nothing to do.
+	})
+
+	return c, nil
+
+}
+
+// controller reconciles resource labels that make claimed resources visible to an APIExport
+// owner. It labels resources in the intersection of `APIBinding.status.permissionClaims` and
+// `APIBinding.spec.acceptedPermissionClaims`.
+type controller struct {
+	queue workqueue.RateLimitingInterface
+
+	kcpClusterClient     kcpclient.Interface
+	apiBindingsIndexer   cache.Indexer
+	dynamicClusterClient dynamic.Interface
+	ddsif                *informer.DynamicDiscoverySharedInformerFactory
+
+	apiBindingsLister apislisters.APIBindingLister
+	listAPIBindings   func(clusterName logicalcluster.Name) ([]*apisv1alpha1.APIBinding, error)
+}
+
+// enqueueAPIBinding enqueues an APIBinding .
+func (c *controller) enqueueAPIBinding(obj interface{}, logSuffix string) {
+	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
+	if err != nil {
+		runtime.HandleError(err)
+		return
+	}
+
+	klog.V(2).Infof("queueing APIBinding %q%s", key, logSuffix)
+	c.queue.Add(key)
+}
+
+// enqueueBindingForResource
+func (c *controller) enqueueBindingForResource(gvr schema.GroupVersionResource, obj interface{}) {
+	// Determine APIBindings that have Permission Claims for the given GVR
+	o, ok := obj.(logicalcluster.Object)
+	if !ok {
+		runtime.HandleError(fmt.Errorf("expected interface: %T but got: %T", o, obj))
+	}
+
+	lc := logicalcluster.From(o)
+
+	apiBindings, err := c.listAPIBindings(lc)
+	if err != nil {
+		runtime.HandleError(err)
+		return
+	}
+
+	for _, binding := range apiBindings {
+		for _, pc := range binding.Spec.AcceptedPermissionClaims {
+			if pc.Group == gvr.Group && pc.Resource == gvr.Resource {
+				key := clusters.ToClusterAwareKey(lc, binding.Name)
+				klog.V(2).Infof("queueing APIBinding %q", key)
+				c.queue.Add(key)
+			}
+		}
+	}
+}
```

Comment:
```suggestion
	for _, binding := range apiBindings {
		for _, pc := range binding.Spec.AcceptedPermissionClaims {
			if pc.Group == gvr.Group && pc.Resource == gvr.Resource {
				c.enqueueAPIBinding(binding, fmt.Sprintf(" because of a permission claim for group %q, resource %q", pc.Group, pc.Resource)
				break 
			}
		}
	}
}
```

Reply:
I don't think that this applies as we are doing this in a new controller, going to resolve it.